cocotb/bus.py

```diff
@@ -32,11 +32,15 @@
 """
 from cocotb.handle import AssignmentResult
 
-def _build_sig_attr_dict(signals):
+
+def _build_sig_attr_dict(signals, name_map=None):
+    if name_map is None:
+        name_map = {}
     if isinstance(signals, dict):
         return signals
     else:
-        return {sig: sig for sig in signals}
+        return {sig: (name_map[sig] if sig in name_map else sig)
```

Comment:
```suggestion
        return {sig: name_map.get(sig, sig)
```
is equivalent and faster